### DISCUSSION TITLE
how to rewrite code with js api?

### DISCUSSION THREAD

**Discussion by scil** [ast-grep/ast-grep#886] [Q&A] [answered]
the doc says `ast-grep's JavaScript API enables you to write JavaScript to programmatically inspect and change syntax trees.` https://ast-grep.github.io/guide/api-usage/js-api.html 

I want to translate every  string literals  from one langugage to another by Google Translator or ChatGPT

I found no example to change the source code.  Would anybody like to share an example?

**Comment by HerringtonDarkholme** [comment, marked as answer]
You can refer to swc's use case 

https://github.com/swc-project/swc/blob/c5053d15746f22f1098e72a0c8f2b7bb8c444d21/packages/helpers/scripts/ast_grep.js#L2
